chia/cmds/param_types.py

```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal, InvalidOperation
+from typing import Any, Callable, Optional, Union
+
+import click
+
+from chia.cmds.units import units
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.bech32m import bech32_decode, decode_puzzle_hash
+from chia.util.config import selected_network_address_prefix
+from chia.util.ints import uint64
+from chia.wallet.util.address_type import AddressType
+
+
+def validate_uint64(
+    value: str,
+    fail_func: Callable[[str, Optional[click.Parameter], Optional[click.Context]], None],
+    param: Optional[click.Parameter],
+    ctx: Optional[click.Context],
+) -> uint64:
+    try:
+        d_value = Decimal(value)
+    except InvalidOperation:
+        fail_func("Value must be a valid uint64 number", param, ctx)
+    if d_value.is_signed():
+        fail_func("Value must be a positive integer", param, ctx)
+    if d_value != d_value.to_integral():
+        fail_func("Value must be an integer", param, ctx)
+    try:
+        u_value = uint64(value)
+    except ValueError:
+        fail_func("Value must be a valid uint64 number", param, ctx)
+    return u_value
+
+
+def validate_decimal_xch(
+    value: str,
+    fail_func: Callable[[str, Optional[click.Parameter], Optional[click.Context]], None],
+    param: Optional[click.Parameter],
+    ctx: Optional[click.Context],
+) -> Decimal:
+    try:
+        d_value = Decimal(value)
+    except InvalidOperation:  # won't ever be value error because of the fee limit check
+        fail_func("Value must be decimal dotted value in XCH (e.g. 0.00005)", param, ctx)
+    if d_value.is_signed():
+        fail_func("Value can not be negative", param, ctx)
+    if not d_value.is_zero() and d_value < Decimal("0.000000000001"):
+        fail_func(
+            "Invalid amount of mojos, too many zeros. Either give zero or at least 1 mojo (0.000000000001 xch).",
+            param,
+            ctx,
+        )
+    return d_value
+
+
+class TransactionFeeParamType(click.ParamType):
+    """
+    A Click parameter type for transaction fees, which can be specified in XCH or mojos.
+    """
+
+    name: str = uint64.__name__  # output type
```

Comment:
chia\cmds\param_types.py:219: error: Cannot override instance variable (previously declared on base class "ParamType") with class variable  [misc]

Reply:
putting ignore for that message

Reply:
So I was bad and didn't provide 'evidence' for my statement.  Turns out I still think it's right, at least in the way we assign it, but it took me a bit to remember this isn't a dataclass so that's not indicating an instance attribute with a default...  Anyways, looking into the error you got I see that upstream indeed did not define this as a class var.  https://github.com/pallets/click/blame/923d197b56caa9ffea21edeef5baf1816585b099/src/click/types.py#L49  So, per them, it is indeed 'supposed' to be an instance var.  One option to avoid the ignore would be an `.__init__()` with a `super()` call to set the instance attribute.

Sorry I didn't look to the next level when I initially commented on this.

Reply:
ok so i should revert it?

Reply:
reveerting change

Reply:
i may revisit this later, let's move forward